L1Trigger/L1THGCal/test/HGCalTriggerGeomTester.cc

```diff
@@ -304,45 +316,80 @@ void HGCalTriggerGeomTester::fillTriggerGeometry(const HGCalTriggerGeometryBase:
     }
     // Loop over cells
     std::cout<<"Filling cells tree\n";
-    for (int izz=0; izz<=1; izz++) 
+    // EE
+    for(const auto& id : info.geom_ee->getValidGeomDetIds())
+    {
+        if(id.rawId()==0) continue;
+        HGCalDetId waferid(id); 
+        int nCells = info.topo_ee->dddConstants().numberCellsHexagon(waferid.wafer());
+        for(int i=0;i<nCells;i++)
+        {
+            HGCalDetId id(ForwardSubdetector(waferid.subdetId()), waferid.zside(), waferid.layer(), waferid.waferType(), waferid.wafer(), i);
```

Comment:
The problem here is that HGCalGeometry::getValidGeomDetIds() returns only ids with HGCalDetId::cell() = 0, so it is necessary to build all the range of cells for each HGCal wafer and do the conversion DetId -> HGCalDetId. Then since for instance HGCalTopology::valid(DetId) is called, the conversion  HGCalDetId -> DetId is done implicitly. 
Improving this is probably beyond the scope of the HGCal trigger code. Here, locally, I don't see clearly how this can be done differently.